marathon.go

```diff
@@ -62,41 +63,39 @@ func (provider *MarathonProvider) loadMarathonConfig() *Configuration {
 			}
 			return ""
 		},
-		"getHost": func(application marathon.Application) string {
-			for key, value := range application.Labels {
-				if key == "traefik.host" {
-					return value
-				}
+		"getWeight": func(task marathon.Task, applications []marathon.Application) string {
+			application := getApplication(task, applications)
+			if application == nil {
+				log.Errorf("Unable to get marathon application from task %s", task.AppID)
+				return "0"
 			}
-			return strings.Replace(application.ID, "/", "", 1)
-		},
-		"getWeight": func(application marathon.Application) string {
-			for key, value := range application.Labels {
-				if key == "traefik.weight" {
-					return value
-				}
+			if label, err := provider.getLabel(*application, "traefik.weight"); err == nil {
```

Comment:
That's too bad.. feels weird (sometimes `*application`, sometimes `application`) :stuck_out_tongue: